lua/lazyvim/plugins/extras/lang/nix.lua

```diff
@@ -0,0 +1,35 @@
+return {
+  {
+    "nvim-treesitter/nvim-treesitter",
+    opts = function(_, opts)
+      opts.ensure_installed = opts.ensure_installed or {}
+      vim.list_extend(opts.ensure_installed, { "nix" })
+    end,
+  },
+  {
+    "williamboman/mason.nvim",
+    opts = function(_, opts)
+      vim.list_extend(opts.ensure_installed, {
+        "nixpkgs-fmt",
+        "nil",
+      })
+    end,
+  },
+  {
+    "neovim/nvim-lspconfig",
+    opts = {
+      servers = {
+        ["nil"] = {},
```

Comment:
From what I'm seeing in [lspconfig docs](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#nil_ls), the name of the server is `nil_ls`. That also means that you don't need to specify it in Mason any more, since it will be installed automatically when declared in `nvim-lspconfig` configuration.

Reply:
Hi @dpetka2001 , could you please provide me the specific code/documentation in `neovim/nvim-lspconfig` in where this happens? I don't want to remove it if I'm not sure, the link you've provided wasn't clear to me.

I'm not sure if changing this to `nil_ls` would work, I see that being the 2nd optional definition in `mason`, but not sure if `nvim-lspconfig` would pick this up. According to lspconfig docs:

```
Default values:

cmd :
{ "nil" }
```
Source:  https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#nil_ls

Let me know if you've manually checked this, if this is verified we can change the syntax to nix_ls instead otherwise I'd like to keep it like this instead.

Reply:
The link to the docs that I provided, specifically say 
> Snippet to enable the language server:
`require'lspconfig'.nil_ls.setup{}`

`"nil"` is the name of the command that will invoke the language server. But in `nvim-lspconfig`, when you're setting up a server, you're passing the name of the language server in the `require`, not the `"cmd"`. You can try it out locally to see how it behaves.

Reply:
:+1: I just manually tried it & it works, doing the change now. Thanks for pointing it out.

Reply:
I'm sorry, but I did not mean that you should remove the `nvim-lspconfig` spec. Maybe I wasn't clear enough. I meant you should do the following
```lua
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        nil_ls = {},
      },
    },
  },
```
and remove the `mason.nvim` spec completely, since all it does at the moment is download the the LSP server, but that is not necessary since every server you specify in `nvim-lspconfig.opts.servers` will be automatically installed.

[Here](https://github.com/LazyVim/LazyVim/blob/bb36f71b77d8e15788a5b62c82a1c9ec7b209e49/lua/lazyvim/plugins/lsp/init.lua#L194-L216) you can find how LazyVim sets up the LSP servers. It checks `nvim-lspconfig.opts.servers` and adds them to `ensure_installed` and passes that table to `mason-lspconfig`, which takes care of the installation and the automatic default handlers for LSP servers.